src/request.cc

```diff
@@ -0,0 +1,275 @@
+// Modeled from parser at: http://www.boost.org/doc/libs/1_62_0/doc/html/boost_asio/example/cpp11/http/server/request_parser.hpp
+// Implementation to parse HTTP requests
+
+#include "request.h"
+
+// Parse some data. The enum return value is good when a complete request has
+// been parsed, bad if the data is invalid, indeterminate when more data is
+// required. The InputIterator return value indicates how much of the input
+// has been consumed.
+std::unique_ptr<Request> Request::Parse(const std::string& raw_request) {
+  std::unique_ptr<Request> parsed_request(new Request());
+  for (char c : raw_request) {
+    parsed_request->consume(c);
+  }
+  return parsed_request;
+}
+
+Request::result_type Request::consume(char input) {
+  m_raw_request.push_back(input);
+
+  switch (m_state) {
```

Comment:
I think each case is supposed to be enclosed in curly braces
https://google.github.io/styleguide/cppguide.html#Loops_and_Switch_Statements